BigQuery/tests/System/ManageModelsTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery\Tests\System;
+
+use Google\Cloud\BigQuery\Dataset;
+use Google\Cloud\BigQuery\Model;
+use Google\Cloud\Core\ExponentialBackoff;
+
+/**
+ * @group bigquery
+ * @group bigquery-model
+ */
+class ManageModels extends BigQueryTestCase
+{
+    /**
+     * @return Model
+     * @throws \Exception
+     */
+    private function createModel()
+    {
+        // Build the create model query
+        $modelId = uniqid(self::TESTING_PREFIX);
+
+        $queryTpl = "CREATE MODEL `%s.%s`
+            OPTIONS (
+              model_type='linear_reg',
+              max_iterations=1,
+              learn_rate=0.4,
+              learn_rate_strategy='constant'
+            ) AS (
+              SELECT 'a' AS f1, 2.0 AS label
+              UNION ALL
+              SELECT 'b' AS f2, 3.8 AS label
+            )";
+
+        // Execute create model query and wait for model to be created
+        $query = sprintf($queryTpl, self::$dataset->id(), $modelId);
+        $config = self::$client->query($query);
+        $results = self::$client->runQuery($config);
+
+        $backoff = new ExponentialBackoff(8);
+        $backoff->execute(function () use ($results) {
+            $results->reload();
+
+            if (!$results->isComplete()) {
+                throw new \Exception();
+            }
+        });
+
+        // Fetch the model and wait for it to be trained
+        $model = self::$dataset->model($modelId);
+
+        $backoff = new ExponentialBackoff(8);
+        $backoff->execute(function () use ($model) {
+            $model->reload();
+            if (!$model->exists()) {
+                throw new \Exception();
+            }
+        });
+
+        // Register the model to be deleted
+        self::$deletionQueue->add(function () use ($model) {
+            $model->delete();
+        });
+
+        return $model;
+    }
+
+    public function testCreatesAndLoadsModelForDataset()
+    {
+        $model = $this->createModel();
+        $this->assertTrue($model->exists());
+    }
+
+    public function testUpdatesModel()
+    {
+        $model = $this->createModel();
```

Comment:
Is there any harm in using the same model from `testCreatesAndLoadsModelForDataset`?

Reply:
I'll make the model a static variable!